cache.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2020 Anas Ait Said Oubrahim
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"container/list"
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type cacheEntry struct {
+	id         string
+	expiryTime int64
+}
+
+type cache struct {
+	mu         sync.Mutex
+	entries    map[string]*prometheus.Gatherer
+	lru        *list.List
+	retention  time.Duration
+	pruneDelay time.Duration
+}
+
+func (c *cache) add(id string, result *prometheus.Gatherer) {
```

Comment:
nitpick: would be nice to have some doc comments like 

```
// Fprint formats using the default formats for its operands and writes to w.
// Spaces are added between operands when neither is a string.
// It returns the number of bytes written and any write error encountered.
func Fprint(w io.Writer, a ...interface{}) (n int, err error) {
```
(see https://blog.golang.org/godoc)

Reply:
done in https://github.com/anas-aso/ssllabs_exporter/pull/11/commits/236ade597cedd42f96bf77eb780a527e2cd5875c